File: leetcode/js/jewels-and-stones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
  let result = 0;
  const set = new Set([...jewels]);

  for (const stone of stones) {
    if (set.has(stone)) {
      result += 1;
    }
  }

  return result;
};

File: leetcode/rust/jewels-and-stones.rs
impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        use std::collections::HashSet;

        let mut result = 0;
        let set: HashSet<char> = jewels.chars().collect();

        for stone in stones.chars() {
            if set.contains(&stone) {
                result += 1;
            }
        }

        result
    }
}
